Style Guide Tutorials (#231)
arduino/docs-content: content/learn/08.contributions/03.arduino-creating-library-guide/arduino-creating-library-guide.md

  delay(3000);
}
```

There are a few differences from the old sketch (besides the fact that some of the code has moved to a library).

First, we've added an #include statement to the top of the sketch. This makes the Morse library available to the sketch and includes it in the code sent to the board. That means if you no longer need a library in a sketch, you should delete the #include statement to save space.

Second, we now create an instance of the Morse class called **morse**:

```arduino
Morse morse(13);
```

> When this line gets executed (which actually happens even before the `setup()` function), the constructor for the Morse class will be called, and passed the argument you've given here (in this case, just 13).

Reviewer: ```suggestion
When this line gets executed (which actually happens even before the `setup()` function), the constructor for the `Morse` class will be called, and passed the argument you've given here (in this case, just 13).
```

Notice that our `setup()` is now empty; that's because the call to `pinMode()` happens inside the library (when the instance is constructed).

> Finally, to call the `dot()` and `dash()` functions, we need to prefix them with **morse**. - the name of the instance we want to use. We could have multiple instances of the Morse class, each on their own pin stored in the _pin private variable of that instance. By calling a function on a particular instance, we specify which instance's variables should be used during that call to a function. That is, if we had both:

Reviewer: ```suggestion
Finally, to call the `dot()` and `dash()` functions, we need to prefix them with **morse**. - the name of the instance we want to use. We could have multiple instances of the `Morse` class, each on their own pin stored in the _pin private variable of that instance. By calling a function on a particular instance, we specify which instance's variables should be used during that call to a function. That is, if we had both:
```

```arduino
Morse morse(13);
Morse morse2(12);
```

then inside a call to `morse2.dot()`, `_pin` would be 12.

If you tried the new sketch, you probably noticed that nothing from our library was recognized by the environment and highlighted in color. Unfortunately, the Arduino software can't automatically figure out what you've define in your library (though it would be a nice feature to have), so you have to give it a little help. To do this, create a file called **keywords.txt** in the Morse directory. It should look like this:

```arduino

> Morse   KEYWORD1
> dash    KEYWORD2
> dot     KEYWORD2

Reviewer: ```suggestion
Morse	KEYWORD1
dash	KEYWORD2
dot	KEYWORD2
```
As explained on the next line, the separator must be a true tab, not spaces:

https://arduino.github.io/arduino-cli/latest/library-specification/#keywordstxt-format

> That's all for now but I'll probably write an advanced library tutorial soon. In the meantime, if you have any problems or suggestions, please post them to the [Software Development forum](https://forum.arduino.cc/c/development/libraries/39).

Reviewer: ```suggestion
If you have any problems or suggestions, please post them to the [Software Development forum](https://forum.arduino.cc/c/development/libraries/39).
```
No point in using up space in the tutorial for speculation about a tutorial that has been hypothetical for a decade.

> For more information, see the [API Style Guide](/content/learn/07.creating-content/arduino-library-style-guide) for information on making a good Arduino-style API for your library.

Reviewer 2: @karlsoderby

> For more information, see the [API Style Guide](/content/learn/07.creating-content/arduino-library-style-guide) for information on making a good Arduino-style API for your library.

Reviewer: ```suggestion
For more information, see the [API Style Guide](/learn/contributions/arduino-library-style-guide) for information on making a good Arduino-style API for your library.
```
Fix link
